tests/integration_tests/test_gas.py

```diff
@@ -42,21 +56,8 @@ def test_gas_call(geth, ethermint):
         ethermint.w3,
         CONTRACTS["BurnGas"])
 
-    # call the contract and get tx receipt for geth
-    geth_gas_price = geth.w3.eth.gas_price
-    geth_txhash = (geth_contract.functions
-                   .burnGas(function_input)
-                   .transact({'from': ADDRS["validator"], "gasPrice": geth_gas_price}))
-    geth_call_receipt = geth.w3.eth.wait_for_transaction_receipt(geth_txhash)
-
-    # repeat the above for ethermint
-    ethermint_gas_price = ethermint.w3.eth.gas_price
-    ethermint_txhash = (ethermint_contract.functions
-                        .burnGas(function_input)
-                        .transact({'from': ADDRS["validator"],
-                                   "gasPrice": ethermint_gas_price}))
-    ethermint_call_receipt = (ethermint.w3.
-                              eth.wait_for_transaction_receipt(ethermint_txhash))
+    geth_call_receipt = get_gas(geth.w3, geth_contract)
+    ethermint_call_receipt = get_gas(ethermint.w3, ethermint_contract)
```

Comment:
The purpose of `get_gas` and `send_signed_tx` is just to reduce the probability of a `TimeExhausted` exception by creating a transaction and waiting for a transaction receipt multiple times, right? If this is going to be a consistent structure to avoid `TimeExhausted` exceptions in the future, it might be a good idea to have a more general recursive function which can be called with another function as input (a function which generates the tx hashes).